airflow/serialization/serialized_objects.py

```diff
@@ -847,10 +847,10 @@ def _serialize_params_dict(cls, params: ParamsDict | dict):
         return serialized_params
 
     @classmethod
-    def _deserialize_params_dict(cls, encoded_params: dict) -> ParamsDict:
+    def _deserialize_params_dict(cls, encoded_params: list) -> ParamsDict:
         """Deserialize a DAG's Params dict."""
         op_params = {}
-        for k, v in encoded_params.items():
+        for k, v in dict(encoded_params).items():
```

Comment:
```suggestion
        for k, v in encoded_params:
```

Reply:
Ah or is this so we can handle both the old and new cases? Converting to dict seems unnecessary to me though. Maybe

```python
if isinstance(encoded_params, collections.abc.Mapping):
    encoded_param_pairs = encoded_params.items()
else:
    encoded_param_pairs = encoded_params
for k, v in encoded_param_pairs:
    ...
```

Reply:
> Ah or is this so we can handle both the old and new cases? 

Yeah, exactly. Yep I'll add that bit of code, makes more clear what we're doing here.